### PR TITLE
Collection of UI improvements and simplifications. 

### PR DESCRIPTION
**Pull request by Xesyto**
- Removed Battle.net login section. Hearthswitcher cannot produce a valid token for it.
- HearthSwitcher automatically attempts to switch back to battle.net configuration when closing.
- When an account has succesfully been created, copies the login information to the Play tab.

- Revised the settings saving behaviour, no longer tries to copy settings from the form that were already Bound in WPF. Saving settings is now a single event handler instead of multiple.
- Seperated code to switch server configuration out of launch game button click events.

### CODE REVIEW

**Comment by Xesyto** [comment]
Added a checkbox to disable auto reverting on closing to the Settings tab. Defaults to off.
Added a button to manually change to battle.net to Settings tab.

File: HearthSwitcher/MainWindow.xaml.cs [hunk 1]
```diff
@@ -200,18 +200,13 @@ public partial class MainWindow : Window
...
-			Properties.Settings.Default.Save();
-		}
-
-		private void emailLogin_TextChanged(object sender, TextChangedEventArgs e) {
-			Properties.Settings.Default.Email = emailLogin.Text;
-			Properties.Settings.Default.Save();
-		}
+	    private void settingChanged(object sender, TextChangedEventArgs e)
+	    {
+            Properties.Settings.Default.Save();
```
Comment (djkaty) [inline HearthSwitcher/MainWindow.xaml.cs]: It doesn't save when you type a character now, or if you click away. You have to press Tab or start typing in another box...

File: HearthSwitcher/MainWindow.xaml.cs [hunk 2]
```diff
@@ -245,6 +245,9 @@ private void settingChanged(object sender, TextChangedEventArgs e)
 				MessageBox.Show("A problem occurred when contacting the server: " + ex.Message);
 			}
 
+		    emailLogin.Text = battletagCreate.Text;
+		    passwordLogin.Password = passwordCreate.Password;
```
Comment (djkaty) [inline HearthSwitcher/MainWindow.xaml.cs]: Did you check that it saves it in the settings once this runs?

File: HearthSwitcher/MainWindow.xaml.cs [hunk 3]
```diff
@@ -266,5 +266,13 @@ private void settingChanged(object sender, TextChangedEventArgs e)
 
 			btnCreateAccount.IsEnabled = true;
 		}
-	}
+
+        private void Mainwindow_Closing(object sender, System.ComponentModel.CancelEventArgs e)
+        {
+            if (!trySwitchOriginal())            
+            {
+                MessageBox.Show("Could not revert Hearthstone to battle.net settings. Closing Hearthstone, restarting and then closing Hearthswitcher should fix this.", "Could not change Hearthstone settings.",MessageBoxButton.OK,MessageBoxImage.Warning);
```
Comment (djkaty) [inline HearthSwitcher/MainWindow.xaml.cs]: If the Hearthstone folder isn't set this causes a 'please set the Hearthstone folder before launching the game' dialog to come up first.

Casing: Battle.net, HearthSwitcher

File: HearthSwitcher/MainWindow.xaml [hunk 4]
```diff
@@ -5,7 +5,7 @@
         xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
         xmlns:local="clr-namespace:HearthSwitcher"
         mc:Ignorable="d"
-        Title="Hearthstone Connect" Height="270" Width="305" VerticalContentAlignment="Center" ResizeMode="CanMinimize">
+        Title="Hearthstone Connect" Height="270" Width="305" VerticalContentAlignment="Center" ResizeMode="CanMinimize" Closing="Mainwindow_Closing">
```
Comment (djkaty) [inline HearthSwitcher/MainWindow.xaml]: Use a vertical margin like in the old Play tab

File: HearthSwitcher/MainWindow.xaml [hunk 5]
```diff
@@ -70,10 +70,14 @@
...
                         </Grid.RowDefinitions>
                         <Label Grid.Row="0" VerticalAlignment="Center">Game location</Label>
-                        <TextBox x:Name="hsPath" Grid.Row="1" Grid.Column="1" Margin="0,0,5,0" TextChanged="settingChanged" Text="{Binding Source={StaticResource Settings}, Path=Default.GamePath}"/>
-                        <Button x:Name="btnFindHearthstone" Content="Browse" Grid.Column="2" Click="btnFindHearthstone_Click" />
+                        <TextBox x:Name="hsPath" Grid.Row="0" Grid.Column="1" Margin="0,0,5,0" TextChanged="settingChanged" Text="{Binding Source={StaticResource Settings}, Path=Default.GamePath}"/>
+                        <Button x:Name="btnFindHearthstone" Grid.Row="0" Content="Browse" Grid.Column="2" Click="btnFindHearthstone_Click" />
+                        <CheckBox x:Name="cbDoNotRevert" Grid.Row="1" Grid.Column="0" Grid.ColumnSpan="3" VerticalAlignment="Center" IsChecked="{Binding Source={StaticResource Settings}, Path=Default.DoNotRevertToBattleNet}" Checked="cbDoNotRevert_Changed" Unchecked="cbDoNotRevert_Changed">Do not revert to Battle.net on closing.</CheckBox>
```
Comment (djkaty) [inline HearthSwitcher/MainWindow.xaml]: Vertical margin

File: HearthSwitcher/MainWindow.xaml [hunk 6]
```diff
@@ -70,10 +70,14 @@
...
+                        <Button x:Name="btnSwitchBattleNet" Grid.Row="2" Grid.Column="0" Grid.ColumnSpan="3" Click="btnSwitchBattleNet_Click">Switch to Battle.net</Button>
```
Comment (djkaty) [inline HearthSwitcher/MainWindow.xaml]: Vertical margin

**Comment by djkaty** [comment]
There is still weird tabby/spacy stuff going on o_O

**Comment by djkaty** [comment]
Browsing to the game folder now doesn't save the selected file unless you click in the text box. Might be related to the settingsChanged bug.

**Comment by Xesyto** [comment]
VS and Reshaper now both set to use tabs on this. Will still produce awkward results every now and then because we have mixed up tabs and spaces going on in the project now.

**Comment by djkaty** [comment]
I agree on the spaces and I'll see if I can get my VS to use tabs for python editing and spaces for C# and do a single commit to replace all tabs with spaces later on. Keep tabs for this PR til it's sorted. I usually use spaces for all my C# work.

**Pull request merged**